### PR TITLE
docs: make reference to stylesheet `val`s clear

### LINKED ISSUE #257: Unable to get the value of `styleSheetText` in Scala 3
I'm having an issue getting the `styleSheetText` when Scala 3 in an example that works for 2.13. For example, given a file like this (which can be ran with scala-cli):

```scala
//> using lib "com.lihaoyi::scalatags:0.12.0"

object Main extends App {
  println(Style.styleSheetText)
}

import scalatags.stylesheet.CascadingStyleSheet
import scalatags.Text.all._

object Style extends CascadingStyleSheet {
  initStyleSheet()

  def example = cls(
    display.flex
  )
}
```

Running this with 2.13.10 you get the following:

```
❯ scala-cli run --scala 2.13.10 .
Compiling project (Scala 2.13.10, JVM)
Compiled project (Scala 2.13.10, JVM)
.Style-example{
  display: flex;
}
```

This is what I'd expect. However, running the same code with Scala 3.2.1 results in an empty output. Same with 3.1.3. 

```
❯ scala-cli run --scala 3.2.1 .

// nothing here
```

It's odd because I do see there are tests for this here:

https://github.com/com-lihaoyi/scalatags/blob/6a93955e5cf8dc204362aecae9ebb5e882f320a6/scalatags/test/src/scalatags/generic/StyleSheetTests.scala#L135-L156

However in practice I can't seem to get this to work.

### PR DESCRIPTION
**Pull request by ckipp01**
In Scala 2 you could use a `def` to define your `cls`s, but in Scala 3
the implementation is only looking for `valDef`s in the file, meaning
that `def`s will no longer work. This makes sure the docs don't mention
using `def`s and instead focus the verbiage on `val`s.

### CODE REVIEW

**Comment by lolgab** [comment]
Do you know if this is an intrinsic limitation of Scala 3 or if it can be fixed? I think the end goal is to have feature parity between Scala 2 and 3

**Comment by ckipp01** [comment]
> Do you know if this is an intrinsic limitation of Scala 3 or if it can be fixed? I think the end goal is to have feature parity between Scala 2 and 3

To be honest I'm not sure without digging in. I _did_ quickly try to add in a check for `DefDef`s here:

https://github.com/com-lihaoyi/scalatags/blob/6a93955e5cf8dc204362aecae9ebb5e882f320a6/scalatags/src-3/scalatags/stylesheet/SourceClasses.scala#L15

But then it screwed up the next line so I didn't go further. My guess is that it's totally doable. So if you'd rather hold off on this, I can try to dig into that and actually fix it correctly if that'd be preferable.

**Comment by lolgab** [comment]
My feeling is more that we should not close the issue if it's an actual issue, instead maybe add more info about what is the actual limitation.
This means that we can probably merge this (after removing "Closes #257" from the PR description, so the original issue remains open).

**Comment by ckipp01** [comment]
> My feeling is more that we should not close the issue if it's an actual issue, instead maybe add more info about what is the actual limitation. This means that we can probably merge this (after removing "Closes #257" from the PR description, so the original issue remains open).

Sure, I put some more info in the issue and I remove the "closed". If I get a bit more free time this week I'll try to dig into a fix.

**Review by lolgab — APPROVED** [review]

**Comment by lolgab** [comment]
Thank you for your contribution @ckipp01 ! :)

**Pull request merged**
